test/conformance/demo-rp/src/pages/api/auth-url.ts

```diff
@@ -0,0 +1,49 @@
+import { generatePkcePair } from '@opengovsg/sgid-client'
+import { setCookie } from 'cookies-next'
+import type { NextApiRequest, NextApiResponse } from 'next'
+import { v4 as uuidv4 } from 'uuid'
+
+import { sgidClient } from '../../lib/sgidClient'
+import { store } from '../../lib/store'
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    store.clear()
+    // sgidClientService.reset()
+    sgidClient.reset()
+
+    // Retrieve the state from the query params
+    let { state } = req.query
+    state = String(state)
+
+    // Generate a session ID
+    const sessionId = uuidv4()
+
+    // Generate a PKCE pair (consisting of code challenge and code verifier)
+    const { codeChallenge, codeVerifier } = generatePkcePair()
+
+    // Generate an authorization URL
+    const { url, nonce } = sgidClient.authorizationUrl({
+      state,
+      codeChallenge,
+    })
+
+    // Store the code verifier, state and nonce in the store
+    store.set(sessionId, { codeVerifier, state, nonce })
+
+    // Set the session ID in the browser cookies
+    setCookie('sessionId', sessionId, {
+      req,
+      res,
+      httpOnly: true,
+      // domain: 'http://localhost:3000',
+      // sameSite: 'none',
```

Comment:
so i found that using `secure: true` results in the test suite failing as the cookie does not get set

my suspicion is that because we are not using a real browser but are just using API calls to simulate browser interaction with the RP, it recognizes localhost as insecure and thus doesn't set the cookie

I've added a code-level comment in [7c69dc3](https://github.com/opengovsg/sgid-client/pull/60/commits/7c69dc329992e00b49f6050dd26659d9af3d87fb) to explain why this change was made